src/Pim/Bundle/EnrichBundle/Controller/FileController.php

```
@@ -79,20 +80,17 @@ public function showAction(Request $request, $filename, $filter = null)
             return $this->renderDefaultImage(FileTypes::MISC, $filter);
         }
 
-        $file = $this->fileInfoRepository->findOneByIdentifier($filename);
-        if (null !== $file) {
-            if (FileTypes::IMAGE === $fileType = $this->fileTypeGuesser->guess($file->getMimeType())) {
-                try {
-                    return $this->imagineController->filterAction($request, $filename, $filter);
-                } catch (NotFoundHttpException $e) {
-                    return $this->renderDefaultImage(FileTypes::IMAGE, $filter);
-                }
-            }
+        $mimeType = $this->getMimeType($filename);
+        $fileType = $this->fileTypeGuesser->guess($mimeType);
 
-            return $this->renderDefaultImage($fileType, $filter);
+        if (FileTypes::IMAGE === $fileType) {
+            try {
+                return $this->imagineController->filterAction($request, $filename, $filter);
+            } catch (NotFoundHttpException $e) {
```

I prefer a c/p than an empty catch block.